### PR TITLE
SRP Workflows - Move VolumeDebugSettings to Core

### PR DESCRIPTION
**Pull request by alex-vazquez**
---
### Purpose of this PR

https://jira.unity3d.com/browse/XPIPELINE-264

First step towards moving the Volume Debug Panel to URP.

Making an interface with the data and implement it on HDRP.

IVolumeDebugSettings <- VolumeDebugSettings<T> <- HDVolumeDebugSettings / (UniversalRenderPipelineVolumeDebugSettings)

---
### Testing status
Testing that the debug panel works in the same way.

### CODE REVIEW

File: com.unity.render-pipelines.core/Runtime/Debugging/VolumeDebugSettings.cs [hunk 1]
```diff
@@ -154,18 +120,26 @@ public static string ComponentDisplayName(Type component)
...
         }
 
-        internal static List<HDAdditionalCameraData> cameras { get; private set; } = new List<HDAdditionalCameraData>();
+        protected static List<T> additionalCameraDatas { get; private set; } = new List<T>();
 
-        internal static void RegisterCamera(HDAdditionalCameraData camera)
+        /// <summary>
+        /// Register the camera for the Volume Debug
+        /// </summary>
+        /// <param name="additionalCamera">The camera with it's additional camera data</param>
```
Comment (arttu-peltonen) [inline com.unity.render-pipelines.core/Runtime/Debugging/VolumeDebugSettings.cs:128]: `it's` -> `its`, or maybe something like `The AdditionalCameraData of the camera to be registered.`. If you change it, do the same for `UnRegisterCamera`.

**Review by arttu-peltonen — APPROVED** [review]
Looks great to me. Pretty minor comments (mostly typos/docs).

**Review by JulienIgnace-Unity — APPROVED** [review]

File: com.unity.render-pipelines.high-definition/Runtime/Debug/DebugDisplay.cs [hunk 2]
```diff
@@ -1646,9 +1646,10 @@ void RegisterVolumeDebug()
             var componentNames = new List<GUIContent>() { new GUIContent("None") };
             var componentValues = new List<int>() { componentIndex++ };
 
-            foreach (var type in VolumeDebugSettings.componentTypes)
+            // TODO @alex.vazquez: Use the same method as VolumeComponentProvider
```
Comment (RSlysz) [inline com.unity.render-pipelines.high-definition/Runtime/Debug/DebugDisplay.cs:1649]: Remaining TODO: checking if it's ok.

File: com.unity.render-pipelines.core/Runtime/Debugging/VolumeDebugSettings.cs [hunk 3]
```diff
@@ -4,26 +4,29 @@
...
 
-        internal int selectedCameraIndex
+        /// <summary>Selected camera index</summary>
+        public int selectedCameraIndex
         {
             get
             {
 #if UNITY_EDITOR
-                if (m_SelectedCamera < 0 || m_SelectedCamera > cameras.Count + 1)
+                if (m_SelectedCamera < 0 || m_SelectedCamera > additionalCameraDatas.Count + 1)
```
Comment (RSlysz) [inline com.unity.render-pipelines.core/Runtime/Debugging/VolumeDebugSettings.cs:26]: I know thiw iw not your change but, `m_SelectedCamera` make me think about a camera object not an index. Should we rename it `m_SelectedCameraIndex`?

**Review by RSlysz — APPROVED** [review]

**Review by RSlysz — APPROVED** [review]

**Review by arttu-peltonen — APPROVED** [review]

**Comment by alex-vazquez** [comment]
@Unity-Technologies/gfx-qa-hdrp please test the Volume debug panel on editor and on standalone build

**Review by JMargevics — APPROVED** [review]
Checked HDRP volume debug view both in editor and built player. No issues ✔

**Pull request merged**